### PR TITLE
PR sur main

### PR DESCRIPTION
**Pull request by pagouillart**

### CODE REVIEW

File: backend/src/controllers/userControllers.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    src: req.src,
+  });
+};
+
+const verifyEmail = async (req, res, next) => {
+  const { email } = req.body;
+  try {
+    const user = await tables.user.findByMail(email);
+    if (user) {
+      res.status(409).send("Email already exists");
```
Comment (Yoann-Fortin) [inline backend/src/controllers/userControllers.js:84]: Status 401 here

File: backend/src/middlewares/validateAdmin.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const adminSchema = Joi.object({
+  user_id: Joi.number().required(),
+});
+const validateAdmin = (req, res, next) => {
+  const adminId = req.body.user_id;
+
+  const { error } = adminSchema.validate({ adminId }, { abortEarly: false });
+
+  if (error) {
+    res.status(422).json({ validationErrors: error.details });
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validateAdmin.js:12]: 403 is right http error here

File: backend/src/middlewares/validateAnecdote.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const artworkId = req.body.artwork_id;
+  const { fact } = req.body;
+
+  const { error } = factSchema.validate(
+    { fact, artworkId },
+    { abortEarly: false }
+  );
+
+  if (error) {
+    res.status(404).json({ validationErrors: error.details });
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validateAnecdote.js:18]: 400 here please

File: backend/src/middlewares/validateArtist.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const { biography } = req.body;
+  const userId = req.body.user_id;
+
+  const { error } = artistSchema.validate(
+    { biography, userId },
+    { abortEarly: false }
+  );
+
+  if (error) {
+    res.status(422).json({ validationErrors: error.details });
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validateArtist.js:18]: 400 here please

File: backend/src/middlewares/validateArtwork.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  // eslint-disable-next-line camelcase
+  const { title, technique, artwork_year, format, alt, user_id_ar } = req.body;
+
+  const { error } = artworkSchema.validate(
+    { title, technique, artwork_year, format, alt, user_id_ar },
+    { abortEarly: false }
+  );
+
+  if (error) {
+    res.status(404).json({ validationErrors: error.details });
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validateArtwork.js:24]: 400 again

File: backend/src/middlewares/validatePassword.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        email: req.email,
+        src: req.src,
+      },
+      process.env.APP_SECRET,
+      { expiresIn: "24h" }
+    );
+    res.cookie("access_token", token, { httpOnly: true });
+    next();
+  } else {
+    res.status(400).send("Invalid Password");
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validatePassword.js:22]: 404

File: backend/src/middlewares/validateUpdateAnecdote.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  fact: Joi.string().required(),
+});
+
+const validateUpdateAnecdote = (req, res, next) => {
+  const { fact } = req.body;
+
+  const { error } = factSchema.validate({ fact }, { abortEarly: false });
+
+  if (error) {
+    res.status(422).json({ validationErrors: error.details });
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validateUpdateAnecdote.js:13]: 400 again

File: backend/src/middlewares/validateUser.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+const validateUser = (req, res, next) => {
+  const { lastname, firstname, email, src, password } = req.body;
+
+  const { error } = userSchema.validate(
+    { lastname, firstname, email, src, password },
+    { abortEarly: false }
+  );
+
+  if (error) {
+    res.status(422).json({ validationErrors: error.details });
```
Comment (Yoann-Fortin) [inline backend/src/middlewares/validateUser.js:20]: 400

File: backend/src/router.js [hunk 9]
```diff
@@ -6,17 +6,68 @@ const router = express.Router();
...
+const validatePassword = require("./middlewares/validatePassword");
+const uploadImage = require("./middlewares/uploadImage");
 
-// Route to add a new item
-router.post("/items", itemControllers.add);
+router.get("/user/:id", userControllers.getUserById);
+router.get("/user", userControllers.getUsers);
+router.post(
+  "/user",
+  userControllers.verifyEmail,
```
Comment (Yoann-Fortin) [inline backend/src/router.js:30]: move this logic in a middleware please

File: backend/src/router.js [hunk 10]
```diff
@@ -6,17 +6,68 @@ const router = express.Router();
...
+  validateUser,
+  hashPassword,
+  userControllers.addUser
+);
+router.post(
+  "/user/login",
+  userControllers.getByMail,
```
Comment (Yoann-Fortin) [inline backend/src/router.js:37]: again please

**Review by Romlion — APPROVED** [review]

File: frontend/src/components/ReturnMainMenuButton/ReturnMainMenuButton.scss [hunk 11]
```diff
@@ -46,24 +46,25 @@
...
   right: 1%;
   top: 1%;
+  font-family: $fontBold;
   cursor: pointer;
   @media screen and (min-width: 700px) {
-    height: 1.5rem;
-    padding: 0 0.5rem;
-    font-size: 0.7rem;
+    height: 1rem;
+    padding: 0 0.2rem;
```
Comment (Yoann-Fortin) [inline frontend/src/components/ReturnMainMenuButton/ReturnMainMenuButton.scss:62]: dead code here

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**